### PR TITLE
Add ServiceHubContextBuilder

### PR DESCRIPTION
**Pull request by Y-Sindo**
* `ServiceManager` uses `ServiceHubContextBuilder` to build `ServiceHubContext`.
* Move `NegotiateAsync` method from `IInternalServiceHubContext` to `ServiceHubContext` abstract class.

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Management/ServiceManager.cs [hunk 1]
```diff
@@ -23,33 +22,25 @@ internal class ServiceManager : IServiceManager
...
 
-        public ServiceManager(IReadOnlyCollection<ServiceDescriptor> services, IServiceProvider serviceProvider, RestClientFactory restClientFactory, IServiceEndpointManager endpointManager, IOptions<ServiceManagerOptions> options)
+        public ServiceManager(IReadOnlyCollection<ServiceDescriptor> services, IServiceProvider serviceProvider, RestClientFactory restClientFactory, IServiceEndpointManager endpointManager)
         {
             _services = services;
             _serviceProvider = serviceProvider;
             _restClientFactory = restClientFactory;
             _endpoint = endpointManager.Endpoints.Keys.First();
             _endpointProvider = endpointManager.GetEndpointProvider(_endpoint);
-            _transportType = options.Value.ServiceTransportType;
```
Comment (wanlwanl) [inline src/Microsoft.Azure.SignalR.Management/ServiceManager.cs:35]: what if cx set ServiceTransportType to Transient, will throw exception or print warning log?

Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.Management/ServiceManager.cs:35]: looks like the ServiceTransportType is useless in the future? plan to remove it?

File: src/Microsoft.Azure.SignalR.Management/ServiceHubContextBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        }
+
+        private Action<IServiceCollection> _configureAction;
+
+        /// <summary>
+        /// Registers an action used to configure <see cref="IServiceManager"/>.
+        /// </summary>
+        /// <param name="configure">A callback to configure the <see cref="IServiceManager"/>.</param>
+        /// <returns>The same instance of the <see cref="ServiceManagerBuilder"/> for chaining.</returns>
+        public ServiceHubContextBuilder WithOptions(Action<ServiceManagerOptions> configure)
```
Comment (wanlwanl) [inline src/Microsoft.Azure.SignalR.Management/ServiceHubContextBuilder.cs:38]: where to use it?

File: src/Microsoft.Azure.SignalR.Management/ServiceManager.cs (hunk 1 above)
Comment (Y-Sindo) [reply src/Microsoft.Azure.SignalR.Management/ServiceManager.cs:35]: `ServiceTransport` is not useless. The behavior is not changed. Just some implementation changes.

File: src/Microsoft.Azure.SignalR.Management/ServiceHubContextBuilder.cs (hunk 2 above)
Comment (Y-Sindo) [reply src/Microsoft.Azure.SignalR.Management/ServiceHubContextBuilder.cs:38]: Users use it.

Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.Management/ServiceHubContextBuilder.cs:38]: Internal class

File: src/Microsoft.Azure.SignalR.Management/ServiceManager.cs (hunk 1 above)
Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.Management/ServiceManager.cs:35]: I see, moved to builder

**Review by wanlwanl — APPROVED** [review]

**Review by vicancy — APPROVED** [review]

**Pull request merged**